### ISSUE TITLE
[FR]

### ISSUE THREAD

**Issue by AntoineBachet** [mlflow/mlflow#5337]
Thank you for submitting a feature request. **Before proceeding, please review MLflow's [Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests) and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.

**Please fill in this feature request template to ensure a timely and thorough response.**

## Willingness to contribute
The MLflow Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature (either as an MLflow Plugin or an enhancement to the MLflow code base)?

- [ ] Yes. I can contribute this feature independently.
- [x] Yes. I would be willing to contribute this feature with guidance from the MLflow community.
- [ ] No. I cannot contribute this feature at this time.

## Proposal Summary

Being able to identify a custom `predict_proba` function from a custom pyfunc model to be able to fully use the "default" evaluators with pyfunc models and not only with sklearn models.
 
## Motivation
- What is the use case for this feature?
Increase the range of models that can use the "default" evaluators when running `mlflow.evaluate`
- Why is this use case valuable to support for MLflow users in general?
It is I believe something easy to unlock that could be used by anyone building their own custom model with pyfunc
- Why is this use case valuable to support for your project(s) or organization?
We would like to use all the evaluators in the group of "default" evaluators with our custom pyfunc classifier.
- Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
This is currently not easy to achieve as we would have to define our own evaluators.

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

## Details

I propose to modify [this function](https://github.com/mlflow/mlflow/blob/af1df139aa107830fbfbc79e515df15f8f6e9f55/mlflow/models/evaluation/default_evaluator.py#L40) in https://github.com/mlflow/mlflow/blob/master/mlflow/models/evaluation/default_evaluator.py to just detect a `predict_proba` within a pyfunc model.

(Use this section to include any additional information about the feature. If you have a proposal for how to implement this feature, please include it here. For implementation guidelines, please refer to the [Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#contribution-guidelines).)

**Comment by henrytdsimmons** [comment]
+1 For this enhancement!

**Comment by paprocki-r** [comment]
yes, please! I would love to use mlflow evaluation for keras multiclassification model

**Comment by sergray** [comment]
Would not it be better to let pass parameter `include_probabilities` to inference API, so model could be called with `predict_proba` instead of `predict`?
